packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -202,6 +202,7 @@ class MDnsVmServiceDiscovery {
       final List<MDnsVmServiceDiscoveryResult> results =
           <MDnsVmServiceDiscoveryResult>[];
       final Set<String> uniqueDomainNames = <String>{};
+      final Set<String> uniqueResultsByDomainName = <String>{};
```

Comment:
The "by" name sounds like it should be a dictionary with a key of domain name mapped to values of unique results.  Can you name it something that describes more how it is different from `uniqueDomainNames`?